packages/components/src/Tabs/Tabs.tsx

```diff
@@ -12,10 +12,24 @@ import { Typography } from "../Typography";
 
 interface TabsProps {
   readonly children: ReactElement | ReactElement[];
+
+  /**
+   * Specifies the tab that should be active on mount
+   */
+  readonly defaultTab?: number;
+
+  /**
+   * Callback that fires when the active tab changes
+   * @param newTabIndex
+   * @param prevTabIndex
+   */
+  onTabChange?(newTabIndex: number, prevTabIndex: number): void;
```

Comment:
oooh what's the useCase for `prevTabIndex` and how confident are we that that's the same value as what would be passed in the `defaultTab`?

thinking about it in this sense

```tsx
const [tab, setTab] = useState(1);

return <Tab defaultTab={tab} onTabChange={setTab}>...</Tab>
```

where the moment that mounts, the `tab` state updates everytime you change the tab therefore making it the previous tab index when `onTabChange` fires 🤔

Reply:
I was thinking about more advanced use-cases like conditional tab logic.

Like "only tabs `1` and `3` would persist. (Why? Who knows?). Or maybe some analytics gathering advanced journey tracking like "users who view tab `2` and then `5` may be trying to find _x_ so present a modal offering to take them there. 

Nothing I've seen so far but it's easy to include and doesn't hurt anything so I figured I'd do it. 

In your example above, `prevTabIndex` would always be the same as the `tab` state at the time the callback is called.

Reply:
Analytics gathering might be a good one. But you still have access to it outside of the component so I'm not sure if we should add it now. especially with this statement

> Nothing I've seen so far but it's easy to include and doesn't hurt anything so I figured I'd do it.

It's one of those easy to add but hella hard to remove

Reply:
Okay, makes sense to me. 

For my understanding, is the concern here mainly that we're adding something that might not get used, so it makes sense to wait till it's needed? I only included it because I couldn't think of a way it would be used to disrupt user flow. I'm still building my comprehension of what our design system standards are in terms of what we expose to our users.

Reply:
Yeah, pretty much. There are a lot of easy things we could do on a generic component but it comes with a cost of maintaining that feature. If we're not sure what it would be used for, i'd rather not have it in there than someone using it for something it wasn't intended to do (since we don't know).

Same with writing a GQL API. There are some easy things you could add on the API as well but if we don't have a solid enough reason to add it, we shouldn't add it as it is much harder to remove if we realize that we don't need it.

Reply:
Okay, I've removed that parameter!